stonesoup/reader/sumo.py

```
@@ -0,0 +1,344 @@
+        # then self.origin will be lat/lon position of the origin of the local Cartesian frame.
+        self.origin = None
+        self.step = 0
+        # Resort to default for sim_start
+        if self.sim_start is None:
+            self.sim_start = datetime.datetime.now()
+
+        self.sumoCmd = [self.sumo_server_path, "-c", self.sumo_cfg_file_path]
+
+        # Raise errors if metadata fields are incorrect
+        if set(self.person_metadata_fields) & {data.name for data in PersonMetadataEnum} != set(
+               self.person_metadata_fields):
+            raise ValueError(f"""Invalid person metadata field(s): {', '.join(str(field) for
+                            field in self.person_metadata_fields if field not in
+                            [data.name for data in PersonMetadataEnum])}""")
+
+        if set(self.vehicle_metadata_fields) & {data.name for data in VehicleMetadataEnum} != set(
+               self.vehicle_metadata_fields):
+            raise ValueError(f"""Invalid vehicle metadata field(s): {', '.join(str(field) for
+                             field in self.vehicle_metadata_fields if field not in
+                             [data.name for data in VehicleMetadataEnum])}""")
+
+    @staticmethod
+    def calculate_velocity(speed, angle, radians=False):
+        if not radians:
+            v_x, v_y = speed * np.sin(np.radians(angle)), speed * np.cos(np.radians(angle))
+        elif radians:
+            v_x, v_y = speed * np.sin(angle), speed * np.cos(angle)
+        return v_x, v_y
+
+    @BufferedGenerator.generator_method
+    def groundtruth_paths_gen(self):
+
+        # Add SUMO_HOME env variable. Use this for server/config-file paths
+        if 'SUMO_HOME' in os.environ:
+            tools = os.path.join(os.environ['SUMO_HOME'], 'tools')
+            sys.path.append(tools)
+            import traci
+        else:
+            sys.exit("Declare environment variable 'SUMO_HOME'")
```

I think this should just raise an error, rather than `sys.exit()`